homeassistant/components/miele/services.yaml

```diff
@@ -23,3 +23,37 @@ set_program:
           max: 99999
           mode: box
       example: 24
+
+set_program_oven:
+  fields:
+    device_id:
+      selector:
+        device:
+          integration: miele
+      required: true
+    program_id:
+      required: true
+      selector:
+        number:
+          min: 0
+          max: 99999
+          mode: box
+      example: 24
+    temperature:
+      required: false
+      selector:
+        number:
+          min: 30
+          max: 300
+          unit_of_measurement: "°C"
+          mode: box
+      example: 180
+    duration:
+      required: false
+      selector:
+        number:
+          min: 1
+          max: 1440
+          unit_of_measurement: "min"
+          mode: box
+      example: 60
```

Comment:
https://www.home-assistant.io/docs/blueprint/selectors/#duration-selector

Wouldn't this be better?

Reply:
I looked at the duration selector but as you cannot disable the seconds field and there is no way to set min and max, I chose the number selector.